src/Admin/AdminConsole/Controllers/OrganizationsController.cs

```diff
@@ -255,29 +255,35 @@ public async Task<IActionResult> Edit(Guid id, OrganizationEditModel model)
             Seats = organization.Seats
         };
 
-        if (organization.PlanType != PlanType.Free && model.PlanType == PlanType.Free && model.Seats > 2)
+        if (model.PlanType.HasValue)
         {
-            TempData["Error"] = "Organizations with more than 2 seats cannot be downgraded to the Free plan";
-            return RedirectToAction("Edit", new { id });
-        }
+            var freePlan = await _pricingClient.GetPlanOrThrow(model.PlanType.Value);
 
-        if (organization.PlanType != PlanType.Free && model.PlanType == PlanType.Free && model.MaxCollections > 2)
-        {
-            TempData["Error"] = $"Organizations with more than 2 collections cannot be downgraded to the Free plan. Your organization currently has {organization.MaxCollections} collections.";
-            return RedirectToAction("Edit", new { id });
-        }
+            if (organization.PlanType != PlanType.Free &&
+                model.PlanType.Value == PlanType.Free &&
+                model.Seats.HasValue &&
+                model.Seats.Value > freePlan.PasswordManager.MaxSeats)
+            {
+                TempData["Error"] = $"Organizations with more than {freePlan.PasswordManager.MaxSeats} seats cannot be downgraded to the Free plan";
+                return RedirectToAction("Edit", new { id });
+            }
 
-        if (organization.PlanType != PlanType.Free && model.PlanType == PlanType.Free)
-        {
-            model.MaxStorageGb = null;
-            model.ExpirationDate = null;
-            model.Enabled = true;
+            if (organization.PlanType != PlanType.Free && model.PlanType.Value == PlanType.Free && model.MaxCollections > freePlan.PasswordManager.MaxCollections)
+            {
+                TempData["Error"] = $"Organizations with more than {freePlan.PasswordManager.MaxCollections} collections cannot be downgraded to the Free plan. Your organization currently has {organization.MaxCollections} collections.";
+                return RedirectToAction("Edit", new { id });
+            }
+
+            if (organization.PlanType != PlanType.Free && model.PlanType.Value == PlanType.Free)
```

Comment:
`organization.PlanType != PlanType.Free && model.PlanType.Value == PlanType.Free` is repeated in each nested `if` statement. I suggest it's moved up to the outer `if` statement, e.g.

```c#
var isDowngradingToFree = model.PlanType.HasValue && organization.PlanType != PlanType.Free && model.PlanType.Value == PlanType.Free;
if (isDowngradingToFree)
{
  // ... your code here
}
```

Reply:
resolved